Tools/RDAP/RDAPQuery/README.md

```diff
@@ -1,5 +1,104 @@
+Author:  Matt Egen 
+
+mattegen@microsoft.com
+
+<a href="https://twitter.com/FlyingBlueMonki?ref_src=twsrc%5Etfw" class="twitter-follow-button" data-show-count="true">Follow @FlyingBlueMonki on Twitter</a>
+
+[![Deploy to Azure](https://aka.ms/deploytoazurebutton)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fgithub.com%2FAzure%2FAzure-Sentinel%2Fblob%2Fmaster%2FTools%2FRDAP%2FRDAPQuery%2Fazuredeploy.json)
+
+With the ever increasing number of new domains on the Internet as well as all of the new Top Level Domains (TLD), it's often hard to know if a user has gone to a potentially malicious new site that has just popped up online.  To help with this, a SOC team or analyst could track for users accessing newly registered domains.  One way to do this is to query the Registration Data Access Protocol (RDAP).  RDAP allows you to access domain name registration data (much like its predecesor the WHOIS protocol does today) but via an API call and with a better, more machine readable structure to the data.  This Azure Function queries an Azure Sentinel environment, finds domain names of interest, and then conducts an RDAP lookup to retrieve information about the domain for investigators and analysts.  There is also an Azure Sentinel Analytic rule that can then alert if evidence of a domain that was registered in the last 30 days should be found.  
+
+Please note:  This version only stores the registration date of the domains successfully resolved, but you could modify it to store more information such as who registered the domain, address information, contact data etc.
+
+Also note: Not all TLD's support RDAP.  The current version of the code does not account for this _except_ to ignore TLD's you specifiy.  A future version of this function is planned to support traditional WHOIS lookups as well if there is enough interest.
+
+## Setup Steps
+When deploying this Azure Function you will need some values to fill in the blanks for the Azure Resource Manager (ARM) Template:
+
+### Function Name
+The name you want to give the Azure Function.  The Default is "RDAPQuery".  A unique string will be attached to this in order to deconflict with any potential pre-existing functions you may have.  For example, the template may create a name like "rdapquerytbdem24sevgdq"
+
+### Azure AD App Permissions
+The Azure Function needs permission to read the Log Analytics Workspace that your Azure Sentinel instance is attached to.  For guidance on creating an Azure AD App Registration, please see [QuickStart: Register App](https://docs.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app).  For this application you will need the following permissions:  

```

Comment:
@FlyingBlueMonkey - Can you please remove locale (en-us) from the link. It should be https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app

Reply:
Will do